In [1]:
import pandas as pd

In [2]:
customer = pd.read_csv('Customer-Churn.csv')

In [3]:
customer.head()

customerID  gender  SeniorCitizen Partner Dependents  tenure PhoneService  \
0  7590-VHVEG  Female              0     Yes         No       1           No   
1  5575-GNVDE    Male              0      No         No      34          Yes   
2  3668-QPYBK    Male              0      No         No       2          Yes   
3  7795-CFOCW    Male              0      No         No      45           No   
4  9237-HQITU  Female              0      No         No       2          Yes   

      MultipleLines InternetService OnlineSecurity  ... DeviceProtection  \
0  No phone service             DSL             No  ...               No   
1                No             DSL            Yes  ...              Yes   
2                No             DSL            Yes  ...               No   
3  No phone service             DSL            Yes  ...              Yes   
4                No     Fiber optic             No  ...               No   

  TechSupport StreamingTV StreamingMovies        Contract PaperlessBilling  \
0          No          No              No  Month-to-month              Yes   
1          No          No              No        One year               No   
2          No          No              No  Month-to-month              Yes   
3         Yes          No              No        One year               No   
4          No          No              No  Month-to-month              Yes   

               PaymentMethod MonthlyCharges  TotalCharges Churn  
0           Electronic check          29.85         29.85    No  
1               Mailed check          56.95        1889.5    No  
2               Mailed check          53.85        108.15   Yes  
3  Bank transfer (automatic)          42.30       1840.75    No  
4           Electronic check          70.70        151.65   Yes  

[5 rows x 21 columns]

In [4]:
x = customer[['tenure']]   # Independent Variable

In [5]:
y = customer[['Churn']]    # Dependent Variable 

In [6]:
from sklearn.linear_model import LogisticRegression

In [7]:
log_model = LogisticRegression()

In [8]:
from sklearn.model_selection import train_test_split

In [9]:
# 35% observation are in test set and 65% observation are in train set.
x_train,x_test,y_train,y_test = train_test_split(x,y,test_size=0.35)

In [10]:
log_model.fit(x_train,y_train)

c:\users\dell1\appdata\local\programs\python\python38\lib\site-packages\sklearn\utils\validation.py:985: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


LogisticRegression()

In [11]:
y_pred = log_model.predict(x_test)  # Predict Value
y_pred

array(['No', 'No', 'No', ..., 'No', 'No', 'Yes'], dtype=object)

In [12]:
y_test.head()  # Actual Value

Churn
2567   Yes
2799    No
3630   Yes
1945    No
293     No

In [13]:
y_pred[0:5]

array(['No', 'No', 'No', 'No', 'No'], dtype=object)

In [14]:
from sklearn.metrics import confusion_matrix

In [15]:
# confusion_matrix(y_test,y_pred)

array([[1754,   86],
       [ 504,  122]], dtype=int64)

The Actual value is 'Yes' but predict value is also 'Yes' so the total is 1754.

The Actual value is 'Yes' but Predict Value is 'No' so the total is 86.

The Actual Value is 'No' but Predict value is 'Yes' so the total is 504.

The Actual Value is 'No' but predict value is also 'No' so the total is 122.

In [17]:
# If we find the accuracy of the model then 

(1817+0)/(1817+0+3+646)

0.7368207623682076